corehq/apps/custom_data_fields/static/custom_data_fields/js/custom_data_fields.js

```diff
@@ -255,6 +255,23 @@ hqDefine('custom_data_fields/js/custom_data_fields', [
                 return profile.serialize();
             });
         };
+        self.profileRequiredForOptions = options.profile_required_for_options || [];
+        let initialRequiredFor;
+        let profileReqiredForMatch;
+        // Check if there is already a user type set as requiring a profile selection and
+        // match to options from UserFieldsView
+        if (options.current_profile_required_for_user_type) {
+            const currentProfileRequiredForList = options.current_profile_required_for_user_type.split(',');
+            profileReqiredForMatch = self.profileRequiredForOptions.find(option =>
+                JSON.stringify(option.value) === JSON.stringify(currentProfileRequiredForList || [])
+            );
+            initialRequiredFor = _.has(profileReqiredForMatch, 'value') ? profileReqiredForMatch.value :
+            (self.profileRequiredForOptions.find(option => option.isDefault)).value || [];
```

Comment:
FYI this is causing an error on staging https://dimagi.atlassian.net/browse/QA-7191

The problem is that option would be `undefined` if it isn't populated for a `UserFieldsView` (So like a `LocationFieldsView` or `ProductFieldsView`

This would solve the issue
```
(self.profileRequiredForOptions.find(function (option) {return option && option.isDefault;}) || {}).value || [];
```

Reply:
Thanks for flagging. One question about the suggestion: Isn't there a chance you'd call `{}.value` ? Like, wouldn't calling `value` on an empty js object throw an error?

Reply:
or maybe it will just return `undefined`

Reply:
ya `{}.value` would return `undefined`. the problem was if option is `undefined` then it `undefined.value` would error